.github/workflows/lint.yaml

```diff
@@ -0,0 +1,18 @@
+name: lint
+on: [push, pull_request]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install ruff
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ruff
+          python -m pip list
+      - name: Run ruff
+        run: ruff --format=github .
```

Comment:
Just to clarify, this just generates a report and does not actually change any files. The action will fail if the linting standards are not met?

Reply:
Correct. `ruff` does have the option to automatically "fix" some subset of errors, though in my experimentation with it some of the autofixes were not correct or otherwise overly aggressive. There are ways to control this in the `ruff` configs (see e.g. `unfixable`), but IMO it'd be easiest to start off without it. We can always revisit this once we gain more experience with the linter in place.